Fixing project initial setup.

olegon opened the pull request:
- I had some issues running `npm run build` and `npm run dev`. This
commit solves these issues on Linux based environment.

Jerenaux commented on package.json:
```diff
@@ -27,7 +27,7 @@
     "test": "mocha ./dist/test.js",
     "watch:server": "nodemon ./dist/server.js",
-    "watch": "webpack-dev-server"
+    "watch:client": "webpack-dev-server"
```
Actually, this command makes `webpack` watch for changes to client **and** server files, so `watch:client` is probably not the right name. However I see the confusion due to the above `watch:server`, which actually makes `nodemon` watch for changes in server dist file specifically (the one generated by webpack), in order to restart the server automatically.

Maybe we could rename these two commands as `nodemon-watch` and `webpack-watch`, or something like that, to emphasize the distinction? 

Let me know if you see what I mean, I let you propose the names. :) Once we are on the same page for that I can merge the PR (and I'll plan to document that somewhere).

Jerenaux replied:
Welcome to the project, by the way!

olegon replied:
Naming things is pretty complicated. What about `watch:nodemon`, `watch:webpack` and just `watch` for the `dev` script? When I want to run some kind of "watching files", I usually search for the *watch* keyword.

If you agree with that, I can fix the `package.json` and update the `README.md`.

About the other changes, they was necessary because Linux is case sensitive. If there is a import for `./MyModule`, *nodejs* will find a `MyModule.js`, not `myModule.js`, or vice-versa.

Jerenaux replied:
The names you propose sound good and should be clear for new users, feel free to commit these changes. :)

The file capitalization is welcome as well and it's good to have a consistent naming convention anyway.

olegon replied:
Done! ;) I've just pushed it again.

olegon committed:
Fixing project initial setup.

- I had some issues running `npm run build` and `npm run dev`. This
commit solves these issues on Linux based environment;
- Renaming npm scripts;
